bundles/http/Citrix/Canvas-LMS/0b9f4444-56d6-4b4e-b108-5624645e71d2.js

```diff
@@ -1,449 +1,398 @@
-              title: announcementValue.title,
-              message: announcementValue.message,
-              enrollme_user_login_id: enrollmentValue.user.login_id,
-              enrollme_role: enrollmentValue.role,
-              posted_at: announcementValue.posted_at,
-              delayed_post_at: announcementValue.delayed_post_at,
-              author_display_name: announcementValue.author.display_name,
-              url: announcementValue.url,
-            };
-            dataStore.save("announcements", announcementBody);
-          }
-        }
-      } else {
-        throw new Error(
-          `Announcement sync failed ${announcementResponse.status}:${announcementResponse.statusText}.`
-        );
-      }
-      tempEnrollmentLink.forEach((element) => {
-        const enrollmentLink = element.split("; ")[1];
-        if (enrollmentLink == 'rel="next"') {
-          enrollmentPageExists = true;
-        }
-      });
-    } while (enrollmentPageExists);
+					for (const announcementValue of announcement) {
+						for (const enrollmentValue of enrollment) {
+							const announcementBody = {
+								id: announcementValue.id,
+								enrollme_id: enrollmentValue.id,
+								enrollme_course_id: enrollmentValue.course_id,
+								title: announcementValue.title,
+								message: announcementValue.message,
+								enrollme_user_login_id: enrollmentValue.user.login_id,
+								enrollme_role: enrollmentValue.role,
+								posted_at: announcementValue.posted_at,
+								delayed_post_at: announcementValue.delayed_post_at,
+								author_display_name: announcementValue.author.display_name,
+								url: announcementValue.url
+							};
+							dataStore.save('announcements', announcementBody);
+						}
+					}
+				} else {
+					throw new Error(
+						`Enrollments sync failed ${enrollmentResponse.status}:${enrollmentResponse.statusText}.`
+					);
+				}
+				tempEnrollmentLink.forEach((element) => {
+					const enrollmentLink = element.split('; ')[1];
 
-    tempAnnouncementLink.forEach((element) => {
-      const announcementLink = element.split("; ")[1];
-      if (announcementLink == 'rel="next"') {
-        announcementPageExists = true;
-      }
-    });
-  } while (announcementPageExists);
+					if (enrollmentLink == 'rel="next"') {
+						enrollmentIncrement = element
+							.split('; ')[0]
+							.replace(/^\<+|\>+$/g, '')
+							.split('?page=')[1]
+							.split('&per_page')[0];
+						enrollmentPageExists = true;
+					}
+				});
+			} while (enrollmentPageExists && !isUpdateAfterAction);
+		} else {
+			throw new Error(
+				`Announcement sync failed ${announcementResponse.status}:${announcementResponse.statusText}.`
+			);
+		}
+
+		tempAnnouncementLink.forEach((element) => {
+			const announcementLink = element.split('; ')[1];
+			if (announcementLink == 'rel="next"') {
+				announcementIncrement = element
+					.split('; ')[0]
+					.replace(/^\<+|\>+$/g, '')
```

Comment:
`<` and `>` don't have to be escaped

Reply:
Of course.
But I meant in this regular expression
```suggestion
					.replace(/^<+|>+$/g, '')
```